optimade/validator/config.py

```diff
@@ -0,0 +1,205 @@
+
+
+_RESPONSE_CLASSES = {
+    "references": ValidatorReferenceResponseMany,
+    "references/": ValidatorReferenceResponseOne,
+    "structures": ValidatorStructureResponseMany,
+    "structures/": ValidatorStructureResponseOne,
+    "info": InfoResponse,
+    "links": ValidatorLinksResponse,
+}
+
+_RESPONSE_CLASSES_INDEX = {
+    "info": IndexInfoResponse,
+    "links": ValidatorLinksResponse,
+}
+
+# This dictionary will eventually be set by getting the fields from response_classes directly
+_PROPERTIES = {
+    "structures": {
+        "MUST": (
+            "id",
+            "type",
+            "structure_features",
+        ),
+        "SHOULD": (
+            "last_modified",
+            "elements",
+            "nelements",
+            "elements_ratios",
+            "chemical_formula_descriptive",
+            "chemical_formula_reduced",
+            "chemical_formula_anonymous",
+            "dimension_types",
+            "nperiodic_dimensions",
+            "lattice_vectors",
+            "cartesian_site_positions",
+            "nsites",
+            "species_at_sites",
+            "species",
+        ),
+        "OPTIONAL": (
+            "immutable_id",
+            "chemical_formula_hill",
+            "assemblies",
+        ),
+    },
+    "references": {
+        "MUST": ("id", "type"),
+        "SHOULD": (),
+        "OPTIONAL": (
+            "last_modified",
+            "immutable_id",
+            "address",
+            "annote",
+            "booktitle",
+            "chapter",
+            "crossref",
+            "edition",
+            "howpublished",
+            "institution",
+            "journal",
+            "key",
+            "month",
+            "note",
+            "number",
+            "organization",
+            "pages",
+            "publisher",
+            "school",
+            "series",
+            "title",
+            "volume",
+            "year",
+            "bib_type",
+            "authors",
+            "editors",
+            "doi",
+            "url",
+        ),
+    },
+}
+
+_UNIQUE_PROPERTIES = ("id", "immutable_id")
+
+_INCLUSIVE_OPERATORS = {
+    DataType.STRING: (
+        "=",
+        "<=",
+        ">=",
+        "CONTAINS",
+        "STARTS WITH",
+        "STARTS",
+        "ENDS WITH",
+        "ENDS",
+    ),
+    DataType.TIMESTAMP: (
+        "=",
+        "<=",
+        ">=",
+        "CONTAINS",
+        "STARTS WITH",
+        "STARTS",
+        "ENDS WITH",
+        "ENDS",
+    ),
+    DataType.INTEGER: (
+        "=",
+        "<=",
+        ">=",
+    ),
+    DataType.FLOAT: (
+        "=",
+        "<=",
+        ">=",
+    ),
+    DataType.LIST: ("HAS", "HAS ALL", "HAS ANY"),
+}
+
+exclusive_ops = ("!=", "<", ">")
+
+_EXCLUSIVE_OPERATORS = {
+    DataType.STRING: exclusive_ops,
+    DataType.TIMESTAMP: exclusive_ops,
+    DataType.FLOAT: exclusive_ops,
+    DataType.INTEGER: exclusive_ops,
+    DataType.LIST: (),
+}
+
+
+class ValidatorConfig(BaseSettings):
```

Comment:
I guess the suggestion of `config.py` instead of a `utils.py` made me think that you might want to customise this? It's obviously a fairly light wrapper, and as its here I don't think we should change it. On the off-chance someone wanted to validate some weird niche implementation, then they could without editing the hardcoded values.

Reply:
Fair enough. It just seems like unnecessary code. But I see your point.